### PR TITLE
Fix matching analyzers from packages to TargetFramework

### LINKED ISSUE #18148: Conditional PackageReference version splitting analyzer presence fails
If a project references two different versions of the same package in two different TargetFrameworks, and the package has added or removed a Roslyn analyzer, the analyzer will be added from _both_ versions of the package, even the one where it's not present.

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFrameworks>net6.0;net472</TargetFrameworks>
    <RootNamespace>two_systemtextjson</RootNamespace>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="System.Text.Json" Version="4.7.0" Condition=" '$(TargetFramework)' == 'net472' " />
    <PackageReference Include="System.Text.Json" Version="6.0.0-preview.4.21253.7" Condition=" '$(TargetFramework)' == 'net6.0' " />
  </ItemGroup>

</Project>
```

```sh-session
two-systemtextjson on master .NET v6.0.100-preview.5.21302.13 🎯 net6.0;net472 
❯ dotnet build  
Microsoft (R) Build Engine version 17.0.0-preview-21302-02+018bed83d for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored S:\work\two-systemtextjson\two-systemtextjson.csproj (in 135 ms).
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
CSC : error CS0006: Metadata file 'C:\Users\raines\.nuget\packages\system.text.json\4.7.0\analyzers\dotnet\cs\System.Text.Json.SourceGeneration.dll' could not be found [S:\work\two-systemtextjson\two-systemtextjson.csproj]
  two-systemtextjson -> S:\work\two-systemtextjson\bin\Debug\net6.0\two-systemtextjson.dll

Build FAILED.

CSC : error CS0006: Metadata file 'C:\Users\raines\.nuget\packages\system.text.json\4.7.0\analyzers\dotnet\cs\System.Text.Json.SourceGeneration.dll' could not be found [S:\work\two-systemtextjson\two-systemtextjson.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.20
```

Because 4.7.0 of that package didn't have an analyzer/source generator, the file isn't found in that package version. The reference is only there because it was found in the other version.

### PR DESCRIPTION
**Pull request by dsplaisted**
Fixes #18148

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs [hunk 1]
```diff
@@ -842,12 +843,10 @@ private void WriteAnalyzers()
 
                         if (targetLibraries == null)
                         {
-                            targetLibraries = _runtimeTarget
-                                .Libraries
-                                .ToDictionary(l => l.Name, StringComparer.OrdinalIgnoreCase);
```
Comment (rainersigwald) [inline src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs:847]: Is it a problem to lose the case-insensitivity?

**Review by rainersigwald — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs (hunk 1 above)
Comment (dsplaisted) [reply src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs:847]: I was wondering the same thing.  I think that it's probably safe to assume that package IDs use consistent casing in the assets file.  @nkolev92 @zkat Does that sound correct?

Comment (nkolev92) [reply src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs:847]: It really depends.
Some sections of the assets file represent user input and those honor the user casing, targets/libraries sections should be ok, but I don't think we have any code that assumes that.

So it really depends on what you are comparing.

Comment (dsplaisted) [reply src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs:847]: @nkolev92 In this case we're comparing `LockFileTargetLibrary` and `LockFileLibrary`.

Comment (nkolev92) [reply src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs:847]: I think it should be safe then. 

You can check that easily by testing with `newtonsoft.json` or `nuget.common`.

**Pull request merged**